scopes/component/isolator/isolator.main.runtime.ts

```diff
@@ -1088,7 +1089,7 @@ export class IsolatorMain {
         const keyName = KEY_NAME_BY_LIFECYCLE_TYPE[dep.lifecycle];
         const entry = dep.toManifest();
         if (entry) {
-          manifest[keyName][entry.packageName] = version;
+          manifest[keyName][entry.packageName] = keyName === 'peerDependencies' ? dep.versionPolicy : version;
```

Comment:
I wonder if the `versionPolicy` might be a snap here, and in that case, you might need to call `snapToSemver`?

Reply:
It is a range, how could it be a snap?

Reply:
ok, I understand, so we need to check that it's a vaild semver when the peer is a snap

Reply:
and what to do if it isn't valid? Throw an error? The versionPolicy is never taken from a calculated field as far as I can tell, so it will only be a snap if a user sets it to be a snap.

Reply:
I think we should cover it with a test and make sure it works as expected.
for example it might have the 0.0.0-hash or being only hash.

Reply:
Works when set through policy

```
          {
            "id": "zoltan.dogfooding/peers/is-peer@94f4f9baef5916517c449c4bf8875dca674be3fa",
            "version": "94f4f9baef5916517c449c4bf8875dca674be3fa",
            "versionPolicy": ">=0.0.0",
            "__type": "component",
            "lifecycle": "peer",
            "source": "auto",
            "componentId": {
              "scope": "zoltan.dogfooding",
              "name": "peers/is-peer",
              "version": "94f4f9baef5916517c449c4bf8875dca674be3fa"
            },
            "isExtension": false,
            "packageName": "@zoltan/dogfooding.peers.is-peer"
          }
```

With bitmap, after I snap, I can't run build again. I get an error:

```
component "zoltan.dogfooding/peers/is-peer@0.0.0" was not found.
this component is a dependency of "zoltan.dogfooding/peers/has-peer-2" and is needed as part of the graph generation
```